sunpy/net/jsoc/jsoc.py

```diff
@@ -483,189 +639,166 @@ def get_request(self, requestIDs, path=None, overwrite=False, progress=True,
+                    break
 
-    def _make_query_payload(self, start_time, end_time, series, notify=None,
-                            protocol='FITS', compression='rice', **kwargs):
-        """
-        Build the POST payload for the query parameters
-        """
+        if wavelength:
+            if not primekey.get('WAVELNTH', ''):
+                if isinstance(wavelength, list):
+                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
+                    wavelength = str(wavelength)
+                else:
+                    wavelength = '{0}'.format(int(np.ceil(wavelength.to(u.AA).value)))
 
-        if protocol.upper() == 'FITS' and compression and compression.lower() == 'rice':
-            jprotocol = 'FITS,compress Rice'
-        elif protocol.upper() == 'FITS':
-            jprotocol = 'FITS, **NONE**'
-        else:
-            jprotocol = protocol
-
-        if not notify:
-            raise ValueError("JSOC queries now require a valid email address "
-                             "before they will be accepted by the server")
-
-        dataset = self._make_recordset(start_time, end_time, series, **kwargs)
-        kwargs.pop('wavelength', None)
-        kwargs.pop('sample', None)
-
-        # Build full POST payload
-        payload = {'ds': dataset,
-                   'format': 'json',
-                   'method': 'url',
-                   'notify': notify,
-                   'op': 'exp_request',
-                   'process': 'n=0|no_op',
-                   'protocol': jprotocol,
-                   'requestor': 'none',
-                   'filenamefmt': '{0}.{{T_REC:A}}.{{CAMERA}}.{{segment}}'.format(series)}
-
-        payload.update(kwargs)
-        return payload
-
-    def _send_jsoc_request(self, start_time, end_time, series, notify=None,
-                           protocol='FITS', compression='rice', **kwargs):
-        """
-        Request that JSOC stages data for download
+            else:
+                error_message = "Wavelength attribute has been passed both as a Wavelength()"\
+                                " and PrimeKey(). Please provide any one of them"\
+                                " or separate them by OR operator."
+                raise ValueError(error_message)
 
-        This routine puts in a POST request to JSOC
-        """
+        else:
+            wavelength = '{0}'.format(primekey.pop('WAVELNTH', ''))
 
-        payload = self._make_query_payload(start_time, end_time, series,
-                                           notify=notify, protocol=protocol,
-                                           compression=compression, **kwargs)
+        if timestr:
+            primekey['TIME'] = timestr
+        if wavelength:
+            primekey['WAVELNTH'] = wavelength
+
+        # Extract and format primekeys
+        pkstr = ''
+        c = drms.Client()
+        si = c.info(series)
+        pkeys_isTime = si.keywords.loc[si.primekeys].is_time
```

Comment:
Could you add some comments to this whole section, I am having some trouble working out what it is doing. Some commented examples of the recordsets being constructed would be useful.